### PR TITLE
IT test for long values

### PR DESCRIPTION
**Pull request by tanvigour**
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #686

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 1]
```diff
@@ -402,6 +404,55 @@ public void testPubSubTTLMetadata() throws Exception {
...
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
+            daprRun.getAppName(),
+            "messages/testinglongvalues",
+            null,
+            HttpExtension.GET, CLOUD_EVENT_LIST_TYPE_REF).block();
+        assertEquals(590518626939830271L, messages.get(0).getData());
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:451]: @tanvigour Why do we have to publish N messages and then check only one message ?

Comment (tanvigour) [reply sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java]: we can check for all values, just added a commit for that

**Comment by mukundansundar** [comment]
@tanvigour  
update workflow files to point to the right dapr commit which has the fix ...

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 2]
```diff
@@ -413,4 +467,13 @@ public void setId(String id) {
       this.id = id;
     }
   }
+
+  public static class ConvertToLong {
+    public Long value;
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:472]: private

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 3]
```diff
@@ -413,4 +467,13 @@ public void setId(String id) {
...
+
+    public void setVal(Long value) {
+      this.value = value;
+    }
+  }
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:477]: Add getter for value, and equals and Dashcode method (Can autogenerate in IntelliJ using Objects.hash etc ).

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 4]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+        60000));
+    // At this point, it is guaranteed that the service above is running and all ports being listened to.
+    if (this.useGrpc) {
+      daprRun.switchToGRPC();
+    } else {
+      daprRun.switchToHTTP();
+    }
+
+    ConvertToLong value = new ConvertToLong();
+    value.setVal(590518626939830271L);
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:423]: Create a Random object to generate random values with `590518626939830271L` as seed. 
Create a Set<ConvertToLong> HashSet, 
Add `590518626939830271L` as the first element. 
Add more random long values upto NUM_MESSAGES size.

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 5]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      for (int i = 0; i < NUM_MESSAGES; i++) {
+        //Publishing messages
+        client.publishEvent(
+            PUBSUB_NAME,
+            LONG_TOPIC_NAME,
+            value,
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:430]: create iterator of the set, and send the next element here.

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 6]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+    }
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
+            daprRun.getAppName(),
+            "messages/testinglongvalues",
+            null,
+            HttpExtension.GET, CLOUD_EVENT_LIST_TYPE_REF).block();
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:450]: ADD an new TypeRef 
`private static final TypeRef<List<CloudEvent<ConvertToLong>>> CLOUD_EVENT_LONG_LIST_TYPE_REF = new TypeRef<>() {};`
use that here .

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 7]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+          Thread.currentThread().interrupt();
+          return;
+        }
+      }
+    }
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:446]:
```suggestion
        final List<CloudEvent<ConvertToLong> messages = client.invokeMethod(
```

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 8]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+          Thread.sleep((long) (1000 * Math.random()));
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+          Thread.currentThread().interrupt();
+          return;
+        }
+      }
+    }
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:443]: Create a new set outside the try block to receive the `actual` values that the Subscriber got for comparison.

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 9]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
+            daprRun.getAppName(),
+            "messages/testinglongvalues",
+            null,
+            HttpExtension.GET, CLOUD_EVENT_LIST_TYPE_REF).block();
+        for (int i = 0; i < NUM_MESSAGES; i++) { 
+          System.out.println("The long value received!!! " +messages.get(i).getData().value);
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:452]: remove log.

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 10]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+
+    try (DaprClient client = new DaprClientBuilder().build()) {
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
+            daprRun.getAppName(),
+            "messages/testinglongvalues",
+            null,
+            HttpExtension.GET, CLOUD_EVENT_LIST_TYPE_REF).block();
+        for (int i = 0; i < NUM_MESSAGES; i++) {
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:451]:
```suggestion
        Assert.assertNotNull(messages);
        for (int i = 0; i < messages.size(); i++) { 
```

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 11]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+      callWithRetry(() -> {
+        System.out.println("Checking results for topic " + LONG_TOPIC_NAME);
+        final List<CloudEvent> messages = client.invokeMethod(
+            daprRun.getAppName(),
+            "messages/testinglongvalues",
+            null,
+            HttpExtension.GET, CLOUD_EVENT_LIST_TYPE_REF).block();
+        for (int i = 0; i < NUM_MESSAGES; i++) { 
+          System.out.println("The long value received!!! " +messages.get(i).getData().value);
+          assertEquals(590518626939830271L, messages.get(i).getData().value);
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:453]: Add it to the `actual`  ConvertToLong Set created above. 
```java
actual.add(message.getData()
```

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 12]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
...
+        }
+      }, 2000);
+    }
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:456]: Add assert comparing two set values 
```java
Assert.assertEquals(expected, actual);
```
Both `expected` and `actual` are of type `Set<ConvertToLong`

File: sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java [hunk 13]
```diff
@@ -402,6 +404,58 @@ public void testPubSubTTLMetadata() throws Exception {
     daprRun.stop();
   }
 
+  @Test
+  public void testlongValues() throws Exception {
```
Comment (mukundansundar) [inline sdk-tests/src/test/java/io/dapr/it/pubsub/http/PubSubIT.java:408]:
```suggestion
  public void testLongValues() throws Exception {
```

**Review by mukundansundar — COMMENTED** [review]
@tanvigour Please address these comments.

**Comment by mukundansundar** [comment]
Tested this locally ... it is working ... 
```
Confirmed listening on port 58606.
Dapr application started.
The long value sent 590518626939830271
== APP == Subscriber got: 590518626939830271
The long value sent 9162974433530815171
== APP == Subscriber got: 9162974433530815171
The long value sent 2803002362178450485
== APP == Subscriber got: 2803002362178450485
The long value sent 1005463900081880990
== APP == Subscriber got: 1005463900081880990
The long value sent 4093312918519397196
== APP == Subscriber got: 4093312918519397196
The long value sent 5939038007520118028
== APP == Subscriber got: 5939038007520118028
The long value sent 7145089449735285569
== APP == Subscriber got: 7145089449735285569
The long value sent 4024458962230509193
== APP == Subscriber got: 4024458962230509193
The long value sent 2837854465277681257
== APP == Subscriber got: 2837854465277681257
The long value sent 562839771211708662
== APP == Subscriber got: 562839771211708662
Checking results for topic testinglongvalues
```

@tanvigour is the test working for you locally ?

**Review by artursouza — APPROVED** [review]

**Pull request merged**